crates/red_knot_python_semantic/src/types.rs

```diff
@@ -1517,6 +1525,14 @@ impl<'db> Type<'db> {
             todo @ Type::Todo(_) => *todo,
             Type::Unknown => Type::Unknown,
             Type::Never => Type::Never,
+            // We consider an instance of any subclass of typing.Any to be an instance of the Any
+            // type.
+            Type::ClassLiteral(ClassLiteralType { class })
+            | Type::SubclassOf(SubclassOfType { class })
+                if class.is_subclass_of_base(db, ClassBase::Any) =>
+            {
+                Type::Any
+            }
```

Comment:
Similarly here, I don't think this is the behavior we want -- an instance of a class that inherits `Any` should still just be an `Instance` type of that class.

Reply:
Reverted